Sources/Units/Unit/Fraction.swift

```diff
@@ -0,0 +1,163 @@
+        // See: https://en.wikipedia.org/wiki/Euclidean_algorithm
+        var latestRemainder = max(a, b)
+        var previousRemainder = min(a, b)
+
+        while latestRemainder != 0 {
+            let tmp = latestRemainder
+            latestRemainder = previousRemainder % latestRemainder
+            previousRemainder = tmp
+        }
+        return previousRemainder
+    }
+}
+
+
+extension Fraction {
+    public static func * (lhs: Self, rhs: Self) -> Self {
+        Self(numerator: lhs.numerator * rhs.numerator, denominator: lhs.denominator * rhs.denominator)
+    }
+
+    public static func / (lhs: Self, rhs: Self) -> Self {
+        Self(numerator: lhs.numerator * rhs.denominator, denominator: lhs.denominator * rhs.numerator)
+    }
+
+    public static func + (lhs: Self, rhs: Self) -> Self {
+        Self(numerator: (lhs.numerator * rhs.denominator) + (rhs.numerator * lhs.denominator), denominator: lhs.denominator * rhs.denominator)
+    }
+
+    public static func - (lhs: Self, rhs: Self) -> Self {
+        Self(numerator: (lhs.numerator * rhs.denominator) - (rhs.numerator * lhs.denominator), denominator: lhs.denominator * rhs.denominator)
+    }
+}
+extension Fraction {
+    public static func * (lhs: Self, rhs: Int) -> Self {
+        lhs * Self(integerLiteral: rhs)
+    }
+
+    public static func / (lhs: Self, rhs: Int) -> Self {
+        lhs / Self(integerLiteral: rhs)
+    }
+
+    public static func * (lhs: Int, rhs: Self) -> Self {
+        Self(integerLiteral: lhs) * rhs
+    }
+
+    public static func / (lhs: Int, rhs: Self) -> Self {
+        Self(integerLiteral: lhs) * rhs
+    }
+
+    public static func + (lhs: Self, rhs: Int) -> Self {
+        lhs + Self(integerLiteral: rhs)
+    }
+
+    public static func - (lhs: Self, rhs: Int) -> Self {
+        lhs - Self(integerLiteral: rhs)
+    }
+
+    public static func + (lhs: Int, rhs: Self) -> Self {
+        Self(integerLiteral: lhs) + rhs
+    }
+
+    public static func - (lhs: Int, rhs: Self) -> Self {
+        Self(integerLiteral: lhs) + rhs
+    }
+}
+
+extension Fraction: ExpressibleByIntegerLiteral {
+    public typealias IntegerLiteralType = Int
+
+    public init(integerLiteral value: Int) {
+        self.init(numerator: value, denominator: 1)
+    }
+}
+
+extension Fraction: SignedNumeric {
+
+    public init?<T>(exactly source: T) where T : BinaryInteger {
+        self.init(integerLiteral: Int(source))
+    }
+
+    public static func *= (lhs: inout Fraction, rhs: Fraction) {
+        lhs = lhs * rhs
+    }
+
+    public var magnitude: Fraction {
+        Self(numerator: abs(numerator), denominator: abs(denominator))
+    }
+    
+    public typealias Magnitude = Self
+
+}
+
+extension Fraction {
+    var asDouble: Double {
+        Double(numerator) / Double(denominator)
+    }
+}
+
+extension Fraction: Comparable {
+    public static func < (lhs: Fraction, rhs: Fraction) -> Bool {
+        return lhs.numerator < rhs.numerator
```

Comment:
I don't think that this is correct - it should be:

```suggestion
        return lhs.asDouble < rhs.asDouble
```

Reply:
You're right. I'll fix this. Instead of `asDouble` I'l just do the int math to create a common denominator and compare them that way.

Reply:
Updated this logic.